src/lib.rs

```diff
@@ -462,6 +471,42 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 			});
 		}
 
+		use hyper::server::conn::http1;
+		use hyper_util::rt::TokioIo;
+		use std::net::SocketAddr;
+		use tokio::net::TcpListener;
+
+		// Start the HTTP server for payjoin
+		let (payjoin_queue_sender, mut payjoin_queue_receiver) = mpsc::channel::<PayjoinRequest>(1);
```

Comment:
@DanGould starting a channel here that is accessed from the payjoin endpoint. the payjoin endpoint will add `PayjoinRequest` to the queue, and wait for response(funding psbt) via different channel from `create_channel_from_pj`.